wordpress-mobile/WordPress-Android
# Issue/10638 material sharing settings

Fixes #10638 

This PR moves Publicize screen to Material Theme and adds Dark Theme support. There are some code changes for cosmetic reasons.

While refactoring is not in the scope of this PR, since I touched those files, I cleaned up/optimized some of the redundant code as suggested by AS, so do not be alarmed by the size of the PR. I suggest turning on "Hide whitespace changes" option during the review.

There is a huge amount of PRs in this feature that covers all the screens within the app, so I'm requesting reviews from a wide variety of people.

![comparison_sharing](https://user-images.githubusercontent.com/728822/70952265-8c1a6500-201a-11ea-88c3-d06c4dbcb9d9.png)

To test:
- Navigate to Publicize screen and confirm it's visual and functional integrity in both Light and Dark themes. The theme can be changed from App Settings.

PR submission checklist:
